### PR TITLE
cabal-install: bootstrap from sources

### PR DESCRIPTION
**Pull request by carlocab**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Let's avoid relying on binaries when they aren't provided by upstream.

### CODE REVIEW

File: Formula/cabal-install.rb [hunk 1]
```diff
@@ -37,11 +37,25 @@ class CabalInstall < Formula
...
   def install
-    resource("bootstrap").stage buildpath
     cabal = buildpath/"cabal"
-    cd "cabal-install" if build.head?
+    if Hardware::CPU.intel?
+      resource("bootstrap").stage buildpath
+    else
+      resource("boostrap").stage do
+        resource("boostrap-json").install Pathname.pwd/"bootstrap"
+        system "boostrap/bootstrap.py", "-d", "bootstrap/darwin-8.10.5.json", "-w", Formula["ghc"].opt_bin/"ghc"
```
Comment (nandahkrishna) [inline Formula/cabal-install.rb:52]:
```suggestion
      resource("bootstrap").stage do
        resource("bootstrap-json").install Pathname.pwd/"bootstrap"
        system "bootstrap/bootstrap.py", "-d", "bootstrap/darwin-8.10.5.json", "-w", Formula["ghc"].opt_bin/"ghc"
```

**Comment by nandahkrishna** [comment]
ARM:
```
File "/private/tmp/cabal-install--bootstrap-20210608-51322-z8iqx6/cabal-Cabal-v3.4.0.0/bootstrap/bootstrap.py", line 193, in install_dep
    raise ValueError(f'Unknown local package {dep.package}')
ValueError: Unknown local package cabal-install-solver
```

**Comment by nandahkrishna** [comment]
The bootstrap version built before the latest push, but the `tar` step failed. Apologies for the multiple force pushes 😅

**Comment by carlocab** [comment]
Thanks @nandahkrishna. No apologies needed.

**Review by MikeMcQuaid — APPROVED** [review]

**Comment by MikeMcQuaid** [comment]
Great work, thanks @carlocab!

**Pull request closed without merging**